core/network/protocols/block_announce_protocol.cpp

```diff
@@ -273,20 +273,27 @@ namespace kagome::network {
                    remote_status.best_block.number);
           self->peer_manager_->updatePeerStatus(peer_id, remote_status);
 
-          // dev mode doesn't have to wait any nodes except itself
-          if (self->app_config_.isRunInDevMode()) {
-            self->babe_observer_->onPeerSync();
-          } else {
+          if (not self->app_config_.isRunInDevMode()) {
             auto self_status = self->createStatus();
             if (not self_status.has_value()) {
               cb(ProtocolError::CAN_NOT_CREATE_STATUS);
               return;
             }
-            if (self_status.value().best_block == remote_status.best_block
-                && self_status.value().roles.flags.authority
-                && remote_status.roles.flags.authority) {
-              self->babe_observer_->onPeerSync();
+
+            if (self_status.value().best_block.number
+                < remote_status.best_block.number) {
+              self->observer_->onRemoteStatus(peer_id, remote_status);
```

Comment:
If key point here to separate block syncing then the same logic in onBlockAnnounce is redundant

Reply:
Condition was removed. thx